src/main/java/com/university/cosmocats/dto/product/CreateProductRequestDto.java

```diff
@@ -0,0 +1,31 @@
+package com.university.cosmocats.dto.product;
+
+import jakarta.validation.constraints.*;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+
+@Data
```

Comment:
I would prefer the builder and value. builer for the simplicity and value for immutabiltiy - 
https://stackoverflow.com/questions/29881135/difference-between-builder-pattern-and-constructor

About immutability - our rule of thumb - if sth could be immutable, we will make it immutable